thunder/distributed/transforms/ddp_v2.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from thunder.core import devices
+from thunder.core import prims
+from thunder.core import utils
+from thunder.core.proxies import DistParallelType
+from thunder.core.trace import from_trace
+from thunder.core.trace import tracectx
+from thunder.core.trace import TraceProvenance
+from thunder.core.transform_common import Transform
+
+if TYPE_CHECKING:
+    from typing import Any
+    from torch.distributed import ProcessGroup
+    from thunder.core.trace import TraceCtx
+
+
+@dataclass(frozen=True)
+class DDPTraceTransform(Transform):
+    process_group: ProcessGroup
+    replicated_params: dict[str, Any]
```

Comment:
IMHO this would be basically `dict[str, torch.nn.Parameter]` or `dict[str, torch.Tensor]`. would you know when/if values are other than those types?

Reply:
Yes, should be torch.nn.Parameter afaik

Reply:
should be always parameters, yes